erica/infrastructure/sqlalchemy/database.py

```diff
@@ -7,11 +9,17 @@
 from erica.config import get_settings
 
 
+def default(obj):
+    if isinstance(obj, decimal.Decimal):
+        return str(obj)
+    raise TypeError
+
+
 def orjson_serializer(obj):
     """
         Note that `orjson.dumps()` return byte array, while sqlalchemy expects string, thus `decode()` call.
     """
-    return orjson.dumps(obj, option=orjson.OPT_SERIALIZE_NUMPY | orjson.OPT_NAIVE_UTC).decode()
+    return orjson.dumps(obj, option=orjson.OPT_SERIALIZE_NUMPY | orjson.OPT_NAIVE_UTC, default=default).decode()
```

Comment:
The only concern I have is the locale. So if we dump a decimal into a string, then which locale will be used for serialize and deserialize? Can you test something like: "1000.0001" or "1000,0001"?

Reply:
Persisting something like "100,001" won't be possible, since a python Decimal value only has a dot as a comma separator. This means the only possible value converted to a string before being persisted would be "100.001" which would be then converted back to Decimal by the worker run in the job.py line 32 using the specific payload for send_est.

Your proposal with:
```
return orjson.dumps(obj, use_decimal=False, option=orjson.OPT_SERIALIZE_NUMPY | orjson.OPT_NAIVE_UTC, default=default).decode()
```

where does the "use_decimal=False" comes from? As far as I see in the official documentation the serialization function only accepts:
```
def dumps(
    __obj: Any,
    default: Optional[Callable[[Any], Any]] = ...,
    option: Optional[int] = ...,
) -> bytes: ...
```

Meaning we can only pass:
- The object to be serialized.
- The default parameter
- The option parameter

I tested the pre-filled tax declaration we have for local development using erica v2 and it worked just as with v1

Reply:
I do not really understand why you had to change something here? 

Should we not have seen that error earlier?

Reply:
We should have seen that error earlier, true. But the only test in erica with the actual engine we use (meaning with the orjson serializer) is TestIntegrationWithDatabaseAndTaxDeclarationJob. And the input est_data does not have any fields which would have a decimal value (the Stmind fields). I have just added one of these fields to that test (and yes it failed without the update on the orjson serializer)

Furthermore I checked the grundsteuer data and you are not persisting any decimal values within the payload column of the entity right?